src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryParser.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.ccd.domain.service.search.elasticsearch;
+
+import static java.util.Optional.ofNullable;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import uk.gov.hmcts.ccd.domain.service.common.ObjectMapperService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
+
+/**
+ * Sample ES query. The parser extracts/sets value for query node.
+ * {
+ * 	"query": {
+ * 		"bool": {
+ * 			"filter": {
+ * 				"match": { "state": "AwaitingPayment"}
+ *                }
+ *            }
+ *       },
+ * 	"sort": {
+ * 		"id": { "order":"asc" }
+ *       }
+ * }
+ */
+class ElasticsearchQueryParser {
+
+    private static final String QUERY_NAME = "query";
+
+    private final ObjectMapperService objectMapperService;
+    private final ObjectNode searchQueryObjectNode;
+
+    ElasticsearchQueryParser(ObjectMapperService objectMapperService, String query) {
+        this.objectMapperService = objectMapperService;
+        this.searchQueryObjectNode = this.objectMapperService.convertStringToObject(query, ObjectNode.class);
+    }
+
+    String extractQueryClause() {
```

Comment:
If you look at the controller `getQuery` method you'll see that we already parse the search request. But then we pass it down to the service as string. I wonder whether now it would be better to pass down the parsed query (Map) . 
If we do that then probably we no longer need this class so we could simplify things a bit. 
Not 100% sure whether that would complicate other areas thought, I leave it to you to decide.